polyply/src/restraints.py

```diff
@@ -81,16 +95,16 @@ def set_restraints(topology, nonbond_matrix):
     for mol_name, mol_idx in topology.distance_restraints:
         distance_restraints = topology.distance_restraints[(mol_name, mol_idx)]
         mol = topology.molecules[mol_idx]
+
+        if set(dict(nx.degree(mol)).values()) != {1, 2}:
+            raise IOError("Distance restraints currently can only be applied to linear molecules.")
```

Comment:
Arguably the path between the two nodes must be strictly linear. Or at least, there can be no intersecting dis_re.

Reply:
see above